src/main/java/org/candlepin/subscriptions/tally/billing/ContractsController.java

```diff
@@ -124,13 +121,29 @@ public Double getContractCoverage(BillableUsage usage) throws ContractMissingExc
   private String getContractMetricId(BillingProvider billingProvider, String productId, Uom uom) {
     TallyMeasurement.Uom measurementUom = TallyMeasurement.Uom.fromValue(uom.toString());
     if (BillingProvider.AWS.equals(billingProvider)) {
-      return tagProfile.awsDimensionForTagAndUom(productId, measurementUom);
+      return getAwsDimension(productId, measurementUom);
     } else if (BillingProvider.RED_HAT.equals(billingProvider)) {
-      return tagProfile.rhmMetricIdForTagAndUom(productId, measurementUom);
+      return getRhmMetricId(productId, measurementUom);
     }
     return null;
   }
 
+  public String getAwsDimension(String productId, TallyMeasurement.Uom uom) {
```

Comment:
`getAwsDimension` and `getRhmMetricId` feel like helper methods that would fit well in one of the swatch-product-configuration classes